src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/NomisCommentService.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.entity.CalculationRequest
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.enumerations.ReleaseDateType
+import uk.gov.justice.digital.hmpps.calculatereleasedatesapi.model.ManualEntrySelectedDate
+import java.time.LocalDate
+
+@Service
+class NomisCommentService {
+
+  fun getNomisComment(
+    calculationRequest: CalculationRequest,
+    isSpecialistSupport: Boolean,
+    approvedDates: List<ManualEntrySelectedDate>?,
+  ): String {
+    val comment = if (isSpecialistSupport) {
+      SPECIALIST_SUPPORT_COMMENT
+    } else if (approvedDates?.isNotEmpty() == true) {
+      MANUAL_ENTERED_DATES_COMMENT
+    } else if (!calculationRequest.reasonForCalculation?.isOther!!) {
```

Comment:
I think this will handle the null case a bit better:

`calculationRequest.reasonForCalculation?.isOther == false`